src/ranged.cpp

```diff
@@ -934,6 +934,14 @@ std::vector<tripoint> target_handler::target_ui( player &pc, target_mode mode,
             idx = -1;
             if( pc.last_target_pos ) {
                 dst = *pc.last_target_pos;
+
+                if( pc.last_target.expired() && pc.has_activity( activity_id( "ACT_AIM" ) ) &&
```

Comment:
If the monster is out of sight but still on the map, pc.last_target.expired() will return true.
I think what you want here is to check for ``pc.last_target.expired() || !pc.sees( pc.last_target.lock().get() )``.
If that works I'll patch it in and merge.